pyanaconda/modules/payloads/payload/dnf/dnf_manager.py

```diff
@@ -494,105 +567,131 @@ def is_package_available(self, package_spec):
-
-        :param exception: an exception
-        :return: True or False
-        """
-        return isinstance(exception, dnf.exceptions.MarkingErrors) \
-            and not exception.error_group_specs \
-            and not exception.error_pkg_specs \
-            and not exception.module_depsolv_errors
+        log.info("Including specs: %s", include_list)
+        for spec in include_list:
+            spec, package_types = get_group_package_types(spec)
+            settings = libdnf5.base.GoalJobSettings()
+            if package_types:
+                settings.set_group_package_types(package_types)
+            self._goal.add_install(spec, settings)
+            self._goal_skip_unavailable.add_install(spec, settings)
 
     def resolve_selection(self):
-        """Resolve the software selection.
-
-        :raise InvalidSelectionError: if the selection cannot be resolved
-        """
-        log.debug("Resolving the software selection.",)
-
-        try:
-            self._base.resolve()
-        except dnf.exceptions.DepsolveError as e:
-            log.error("The software couldn't be resolved!\n%s", str(e))
-
-            message = _(
-                "The following software marked for installation has errors.\n"
-                "This is likely caused by an error with your installation source."
-            )
-
-            raise InvalidSelectionError(message + "\n\n" + str(e).strip()) from None
-
-        log.info("The software selection has been resolved (%d packages selected).",
-                 len(self._base.transaction))
+        """Resolve the software selection."""
+        report = ValidationReport()
+        messages = []
+
+        log.debug("Resolving the software selection.")
+        self._transaction = self._goal.resolve()
+
+        problems = self._transaction.get_problems()
+        if problems != libdnf5.base.GoalProblem_NO_PROBLEM:
+            # Store the resolve logs from this transaction. If we can resolve the transaction with
+            # skip_if_unavailable=True, these will be just warnings, otherwise, errors.
+            for message in self._transaction.get_resolve_logs_as_strings():
+                messages.append(message)
+
+            # The list might not be exhaustive, but these errors definitely shouldn't be skipped.
+            critical_errors = libdnf5.base.GoalProblem_SOLVER_ERROR | \
+                libdnf5.base.GoalProblem_SOLVER_PROBLEM_STRICT_RESOLVEMENT | \
+                libdnf5.base.GoalProblem_UNSUPPORTED_ACTION | \
+                libdnf5.base.GoalProblem_MULTIPLE_STREAMS | \
+                libdnf5.base.GoalProblem_MODULE_SOLVER_ERROR_DEFAULTS | \
+                libdnf5.base.GoalProblem_MODULE_SOLVER_ERROR_LATEST | \
+                libdnf5.base.GoalProblem_MODULE_SOLVER_ERROR | \
+                libdnf5.base.GoalProblem_MODULE_CANNOT_SWITH_STREAMS
+
+            # If we didn't already run with skip_unavailable=True and the problems are not from the
```

Comment:
Does this override the user or system configuration, even though the config explicitly has skip_unavailable = False?

Reply:
This behavior is actually quite necessary on RHEL for example, where for whatever reason the insights-client package is usually not available until later in thy cycle & we enable Insights by default as was requested.

So with this feature you can skip the missing Insights package and finish the installation. If we aborted installation due to the missing package, it could break some RHEL testing. If we silently ignored it, we could miss Insights missing at GA or similar.